### PR TITLE
yubikey-manager-qt: build using Qt mkDerivation

### PR DESCRIPTION
**Pull request by peterhoeg**
##### Motivation for this change

Further to https://github.com/NixOS/nixpkgs/issues/65399

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @avdv

### CODE REVIEW

**Comment by avdv** [comment]
> I'm intrigued by where the python code is in this application.

I think it gets packaged into the binary as a resource. Depending on how the app is started (out of the build directory or not) it gets loaded via the resource or directly from the file system: https://github.com/Yubico/yubikey-manager-qt/blob/master/ykman-gui/main.cpp#L58-L70

File: pkgs/tools/misc/yubikey-manager-qt/default.nix [hunk 1]
```diff
@@ -27,30 +24,29 @@ stdenv.mkDerivation rec {
     sha256 = "087ms9i0n3rm8a0hvc4a2dk3rffbm6rmgz0m8gbjk6g37iml6nb7";
   };
 
-  nativeBuildInputs = [ wrapQtAppsHook python3.pkgs.wrapPython qmake ];
-
   postPatch = ''
-    substituteInPlace ykman-gui/deployment.pri --replace '/usr/bin' "$out/bin"
+    substituteInPlace ykman-gui/deployment.pri \
+      --replace /usr/bin $out/bin
```
Comment (avdv) [inline pkgs/tools/misc/yubikey-manager-qt/default.nix]: Why did you change this line? Did you ran some linter tool? But then, why didn't you also change a similar line for the `substituteInPlace` call at line 54?

Usually, I am in favor of proper quoting, to be safe. But at least it should be consistent in the same file.

**Review by avdv — COMMENTED** [review]
LGTM, I build it locally and it worked for me.

**Comment by avdv** [comment]
> Though I'm wondering this, does the shebang get corrected before it gets embedded?
> If not that would introduce an impurity relying on python being in the environment.

I don't think the shebang is in any way relevant, since the file is never unpacked on disc for a shell to consume. It is directly called by the embedded interpreter.

**Comment by peterhoeg** [comment]
Someone already did this. No longer needed.
